.eslintrc.js

```diff
@@ -30,9 +30,13 @@ module.exports = {
           '**/__tests__/**/*.js',
           '**/.storybook/**/*.*',
           '**/*{.,_}{test,spec}.{ts,tsx}',
+          'app/__mocks__/**/*',
```

Comment:
what will this do?

Reply:
eslint rule `import/no-extraneous-dependencies` will throw error, if we are importing any package which is not defined in `dependencies` field in `package.json`.

That means if we are importing `jest` or `@testing-library/react` in our mocks file or test file, eslint will throw error. To fix these we have to tell to `eslint` that those files are not source code. 

That's why I have added `app/__mocks__/**/*`  and `app/mocks/**/*` folders

Reply:
right now we are getting `import/no-extraneous-dependencies` for files inside these `mock` folders, but we are not getting it inside the components folder , do u know why?

Reply:
`import`no-extraenous-dependencies` will throw error only when you are importing a package which is not mentioned in as dependencies in `package.json` .

For example if you import from `@remix-run/node` it will not throw error since it is defined as `dependencies` whereas if you import from `@testing-library/react` it will throw since `@testing-library/react` is defined as `devDependencies`. 
![image](https://github.com/Real-Dev-Squad/website-calendar/assets/52579256/e227b172-cba8-4c34-ad57-64f657fbb55b)

Reply:
Yes i understand what you are saying, but if you check `/components/common/anycomponent/index.test.tsx`, it has  
imported `@testing-library/react`, but it doesn't show this error. but the `/components/common/anycomponent/index.tsx`, shows that error thats what i was asking.

Reply:
I think that's because there is an preexisting rule in `eslint.js` with ` '**/*{.,_}{test,spec}.{ts,tsx}',`  (https://github.com/Real-Dev-Squad/website-calendar/blob/3615fd7a63c3e1d81abd563a9594576d4980b902/.eslintrc.js#L21-L32) . Because of that rule any file with `*.test.ts` or `*.test.tsx` will be considered as dev dependency